llvm/utils/TableGen/DecoderEmitter.cpp

```diff
@@ -935,7 +935,7 @@ void DecoderEmitter::emitTable(formatted_raw_ostream &OS, DecoderTable &Table,
       unsigned Shift = 0;
       do {
         OS << ", " << (unsigned)*I;
-        Value += (*I & 0x7f) << Shift;
+        Value += ((uint64_t)(*I & 0x7f)) << Shift;
```

Comment:
Not necessary. We can confirm (*I & 0x7f) is safe to do shift.

Reply:
+1

Reply:
ok

Reply:
Sorry, I think the (uint64_t) in needed, because `Value` is uint64_t and `Shift` may larger than 32.

Reply:
i see. there are no pending issues here, can you approve & merge request?